azuread/resource_service_principal.go

```diff
@@ -44,6 +45,11 @@ func resourceServicePrincipal() *schema.Resource {
 				Computed: true,
 			},
 
+			"app_role_assignment_required": {
```

Comment:
Could we move this above the computed properties?

Reply:
Done. What about the `oauth2_permissions` and `tags`, would you like me to move those above the computed properties as well, for consistency's sake?